lld/COFF/Driver.cpp

```diff
@@ -858,46 +858,6 @@ static std::string createResponseFile(const opt::InputArgList &args,
   return std::string(data.str());
 }
 
-enum class DebugKind {
-  Unknown,
-  None,
-  Full,
-  FastLink,
-  GHash,
-  NoGHash,
-  Dwarf,
-  Symtab
-};
-
-static DebugKind parseDebugKind(const opt::InputArgList &args) {
-  auto *a = args.getLastArg(OPT_debug, OPT_debug_opt);
-  if (!a)
-    return DebugKind::None;
-  if (a->getNumValues() == 0)
-    return DebugKind::Full;
-
-  DebugKind debug = StringSwitch<DebugKind>(a->getValue())
-                        .CaseLower("none", DebugKind::None)
-                        .CaseLower("full", DebugKind::Full)
-                        .CaseLower("fastlink", DebugKind::FastLink)
-                        // LLD extensions
-                        .CaseLower("ghash", DebugKind::GHash)
```

Comment:
I think all options like 'full', 'fastlink', 'ghash', 'noghash' should just act in the same way and enable full debug with GHASH merging. We have GHASH now for several years now and it works well, I think we could disable support for the traditional, non-multithreaded, type merging. Maybe warn on 'noghash' like we do for 'fastlink' and say it's no longer supporter. +@rnk

Reply:
@aganea I guess that sounds reasonable, although I'd prefer not to take on that kind of a change in behaviour as part of this patchset, it's tricky enough as it is (even if I guess it would make the code a bit simpler in some places) - I'd leave that to someone more familiar with PDB/ghash overall.